### ISSUE TITLE
xSCCMSQLSetup 

### ISSUE THREAD

**Issue by PowerShellCrack** [dsccommunity/ConfigMgrCBDsc#15]
### README.md
 - Could there be the ability to have the DSC during build run a Save-Module any additional requiredmodules for offline support. For instance _xSCCMSQLSetup_ requires the **SqlServerDsc version 13.5.0**. I would have to drill into each module to find this. 

	eg. 
`Save-module sqlserverdsc -Path U:\DSC\ConfigMgrCBDsc\output\RequiredModules -RequiredVersion 13.5.0`

 in README all examples go nowhere

### xSCCMSQLSetup

- InstallSqlDataDir --> README states default is **C:**, so I assumed if I change it to **E:** it would install on the VM E drive, yet the code states it needs just a letter (eg. **E**). The README should be changed or add a Filepath validation in the module to make sure **E:\, E: or E** can be used. 

-  UpdateEnabled--> Great feature nut my VM's are offline. Any way there could be an autocheck for an updates folder in the SQLInstallPath and add the param to the installer (eg. _/UpdateSource <path to updates folder>_)

	Reference: _https://docs.microsoft.com/en-us/sql/database-engine/install-windows/install-sql-server-from-the-command-prompt?view=sql-server-ver15_

- I don't see the ability to add features to SQL. I'm installing Sql 2014 and SSMS is native to the module (line 259: Features            = 'SQLENGINE,RS,CONN,BC,SSMS,ADV_SSMS' ), However I would like to add Reporting server and **not** the SSMS that's native but a newer version from the external installer (18.5) and not to do both. I could also see where SSMS is sometime not used directly on the server but would be on a Admin workstation so an option.  Adding features parameter would be useful. 

 - Also if I run SQL 2017, RS feature is no longer available and would have to be installed separate. 

Or do you think I should just use the **SqlServerDSC** module?

**Comment by rchristman89** [comment]
I was trying to not require a specific version of the SqlServerDsc Module but was running into some issues. That is something I can look into again. There isn't really a way to download the module in a config since if that module is used in the config it has to be present when it is compiled. 

*xSccmSqlSetup*
- InstallSqlDataDir - I modeled it after the resource and it accepts a string the only validation is to remove the trialing slash if there is one but the resource does that one. I may be missing where it says it only needs to be the letter. 
- UpdateEnabled - I can add this but it will increase the config size because I will have to put logic if UpdateSource is provided it uses a different config with different parameters. 
- I can change the Features to a default and provide those defaults. I picked those because they are what SCCM needs that is an easy fix. But the idea with the composite is to automate the setup needed. If there are more requirements using the SqlServerDsc module would be recommended. 
- For .NetFramework 4.5, that is done in the xSccmPreReqs composite resource. I would say that should be ran prior to the SQL setup.
